### PR TITLE
The level of verbosity of messages printed to the console can now be changed

### PR DESCRIPTION
**Pull request by transiient**
As suggested in bug #1636626, the level of verbosity for messages printed to the console can now be changed with the command line argument `--debugLevel <0-2>`.

* 0 - Show only Fatal and Critical messages. These are always shown.
* 1 - Show above + Warning messages. This is the default option chosen when `--debugLevel` isn't set or when it's set to a non-integer value.
* 2 - Show above + Debug messages.

Anything above 2 won't make a difference, as there are only 3 levels of debug messages. A negative integer will, in theory, set `debugLevel` to 0, as Fatal and Critical never need to check what the argument is set to.

This is my first PR on a large project so I hope I'm doing this right. I've tested it and it appears to be working as it should. Some logging messages may need updating to split them better between errors, warnings, etc.

### CODE REVIEW

**Comment by vhexs** [comment]
Wow, I'm either too tired to read this properly right now or I'm an idiot - it seems I forgot to update the debugLevel variable in logging.cpp. I'll have a look at this tomorrow, maybe I didn't do enough testing and I missed something.

**Comment by daschuer** [comment]
Thank you for this PR. I hope I will find time to test it soon. 

In order to merge this to master, we need your permission. Please sign: 
https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

**Comment by vhexs** [comment]
All done!

File: src/util/cmdlineargs.cpp [hunk 1]
```diff
@@ -52,12 +53,23 @@ bool CmdlineArgs::Parse(int &argc, char **argv) {
         } else if (argv[i] == QString("--timelinePath") && i+1 < argc) {
             m_timelinePath = QString::fromLocal8Bit(argv[i+1]);
             i++;
+        } else if (argv[i] == QString("--debugLevel") && i+1 < argc) { // Set cmdline message verbosity
+            bool isInt; // Scoped variable, will be destroyed at the end of this block @vhexs
```
Comment (uklotzde) [inline src/util/cmdlineargs.cpp]: Please remove this comment. There is nothing special about a simple local variable of type bool that has neither a constructor nor a destructor. It is just kicked off the stack when leaving the current scope. Comments should only be used for revealing something that is not directly visible from the code.

On the other hand I prefer to initialize such variables with a default value, even it should be set immediately by the following function call. You never know what might go wrong inside this function.

File: src/util/cmdlineargs.cpp [hunk 2]
```diff
@@ -52,12 +53,23 @@ bool CmdlineArgs::Parse(int &argc, char **argv) {
...
+            m_debugLevel = QString::fromLocal8Bit(argv[i+1]).toInt(&isInt); // Try conversion, if it's not an int
```
Comment (uklotzde) [inline src/util/cmdlineargs.cpp]: This comment is redundant, too, Your code is readable, which is good :+1:

File: src/util/logging.cpp [hunk 3]
```diff
@@ -87,37 +88,45 @@ void MessageHandler(QtMsgType type,
         Logfile.open(QIODevice::WriteOnly | QIODevice::Text);
     }
 
+    debugLevel = CmdlineArgs::Instance().getDebugLevel(); // Get verbosity level
+
     switch (type) {
-    case QtDebugMsg:
-        fprintf(stderr, "Debug %s", ba.constData());
+    // debugLevel is set to 1 by default, can be changed with the cmdline
```
Comment (uklotzde) [inline src/util/logging.cpp]: As I already mentioned, this intention can be captured by defining a constant for the default value.

File: src/util/cmdlineargs.cpp [hunk 4]
```diff
@@ -12,6 +12,7 @@ CmdlineArgs::CmdlineArgs()
       m_developer(false),
       m_safeMode(false),
       m_settingsPathSet(false),
+      m_debugLevel(1),
```
Comment (uklotzde) [inline src/util/cmdlineargs.cpp]: Our anonymous default value again, see below ;)

File: src/util/cmdlineargs.cpp [hunk 5]
```diff
@@ -52,12 +53,23 @@ bool CmdlineArgs::Parse(int &argc, char **argv) {
...
+        } else if (argv[i] == QString("--debugLevel") && i+1 < argc) { // Set cmdline message verbosity
+            bool isInt; // Scoped variable, will be destroyed at the end of this block @vhexs
+            m_debugLevel = QString::fromLocal8Bit(argv[i+1]).toInt(&isInt); // Try conversion, if it's not an int
+                                                                            // then the if statement below will
+                                                                            // set m_debugLevel back to its
+                                                                            // default value of 1
+            if (!isInt) {
+                fputs("\ndebugLevel argument wasn't a number! Mixxx will only output\n\
+warnings and errors to the console unless this is set properly.\n", stdout);
+                m_debugLevel = 1;
```
Comment (vhexs) [inline src/util/cmdlineargs.cpp]: Sorry for the noobish question, but where would I initialize the constant?

Comment (uklotzde) [inline src/util/cmdlineargs.cpp]: Logging should define both the range of supported values (min/max) as well as the default value. Use a public static const member in Logging for this purpose. A simple integer can directly be initialized with its declaration in the header file:
static const int kMinDebugLevel = 0;
static const int kMaxDebugLevel = 2;
static const int kDefaultDebugLevel = 1;

Wether to use an integer or an enum? Well, I'm fine with the integer if you avoid to use the magic numbers 0,1,2 in the code ;)

Comment (vhexs) [inline src/util/cmdlineargs.cpp]: Ah, I see what you mean now. I'll try using integers as I'm not very good with enums.

**Comment by vhexs** [comment]
I've added default and minimum values to `logging.h`, and removed the unnecessary comments I added, however I wasn't sure whether or not to put them in the logging class so I left them out. I also set debug message prefixes back to how they were before I changed them, as I didn't realise it printed the current thread after the prefix (oops - I'll check next time).

**Comment by daschuer** [comment]
Thank you. LGTM!

**Pull request merged**